### PR TITLE
Hotkeys: initial implementation for this addon

### PR DESCRIPTION
**Pull request by humitos**
Initial implementation for the "hotkeys" addon:

- allow to enable disable the whole addon or any other specific addon (e.g. only disable "search")
- allow to customize each particular hotkey per addon (uses `config.addons.hotkeys.doc_dif.trigger`)
- simple syntax to define the trigger. Example `Ctrl + k` (thanks to https://www.npmjs.com/package/keyboard-event-to-string)

This also allows us to handle _all the hotkeys_ in a single place, which will easier to develop, disccuss, maintain, and handle.

* Related: https://github.com/readthedocs/addons/issues/117
* Related: https://github.com/readthedocs/addons/pull/119

### CODE REVIEW

File: src/hotkeys.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      elems[0].requestUpdate();
+    }
+
+    for (const elem of elems) {
+      elem.loadConfig(config);
+    }
+  }
+
+  static isEnabled(config) {
+    return config.addons && config.addons.hotkeys.enabled;
```
Comment (agjohnson) [inline src/hotkeys.js:88]: This is another place that a try/catch helps, as `hotkey` could not be defined.

**Review by agjohnson — APPROVED** [review]
Interesting idea. I'm up for testing this approach, it seems wise for us consolidate the key handling logic.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ repos:
     hooks:
       - id: prettier
         args: ["--write", "src/**", "*.js", "*.json"]
-
+        exclude: "^dist/.*$"
```
[the remark this answers is not in the document]
Comment (humitos) [reply .pre-commit-config.yaml:7]: Yeah, I'm not sure how to share this information. I will think about this and see if I can standardize it.

**Comment by humitos** [comment]
I'm merging this PR, but I'm opening an issue to talk about the handling exceptions because I'm not sure what's the way to do it or a good pattern to follow yet. I understand the problem, but I'd like to not overload the whole code of try/catch and I'm not sure what's a good way to do that.

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (agjohnson) [reply .pre-commit-config.yaml:7]: Not a huge deal, prettier doesn't run with many options either way. Just noting it for now I suppose

**Comment by agjohnson** [comment]
Roger, that makes sense. I mostly hit this issue trying to add more testing in #73, as the configuration object was missing entirely.

**Pull request merged**